sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -88,3 +90,21 @@ def test_extract_intro():
             result,
             'This is the description of the example which goes on and on')
         assert_false('second paragraph' in result)
+
+
+def test_ipy_notebook():
+    """Test that written ipython notebook file corresponds to python object"""
+    with tempfile.NamedTemporaryFile('w+') as f:
+        example_nb = gnb.Notebook(f.name, os.path.dirname(f.name))
+        blocks = sg.split_code_and_text_blocks('tutorials/plot_parse.py')
+
+        for blabel, bcontent in blocks:
+            if blabel == 'code':
+                example_nb.add_code_cell(bcontent)
+            else:
+                example_nb.add_markdown_cell(sg.text2string(bcontent))
+
+        example_nb.save_file()
+
+        f.flush()
+        assert_equal(gnb.json.load(f), example_nb.work_notebook)
```

Comment:
It would be good to directly import json, and not use the one that is inherited from gnb. The reason is that, to the reader that doesn't know the entire codebase, it is not clear at all that "gnb.json" is the json module from the stdlib.